hazelcast-client-legacy/src/test/java/com/hazelcast/client/map/ClientMapTest.java

```diff
@@ -275,6 +276,33 @@ public void entryEvicted(EntryEvent<String, String> event) {
     }
 
     @Test
+    public void testAsyncSet() throws Exception {
+        IMap<String, String> map = createMap();
+        fillMap(map);
+        Future<Void> future = map.setAsync("key3", "value");
+        future.get();
+        assertEquals("value", map.get("key3"));
+    }
+
+    @Test
+    public void testAsyncSetWithTtl() throws Exception {
+        IMap<String, String> map = createMap();
+        final CountDownLatch latch = new CountDownLatch(1);
+        map.addEntryListener(new EntryEvictedListener<String, String>() {
+            public void entryEvicted(EntryEvent<String, String> event) {
+                latch.countDown();
+            }
+        }, true);
+
+        Future<Void> future = map.setAsync("key", "value1", 3, TimeUnit.SECONDS);
+        future.get();
+        assertEquals("value1", map.get("key"));
+
+        assertTrue(latch.await(10, TimeUnit.SECONDS));
```

Comment:
Okay. Let me make the changes that you recommend - these tests were largely
taken from their putAsync equivalents.
On 29 Dec 2015 4:08 pm, "Peter Veentjer" <notifications@github.com> wrote:

> In
> hazelcast-client-legacy/src/test/java/com/hazelcast/client/map/ClientMapTest.java
> <https://github.com/hazelcast/hazelcast/pull/7129#discussion_r48526150>:
>
> > +
> > +    @Test
> > +    public void testAsyncSetWithTtl() throws Exception {
> > +        IMap<String, String> map = createMap();
> > +        final CountDownLatch latch = new CountDownLatch(1);
> > +        map.addEntryListener(new EntryEvictedListener<String, String>() {
> > +            public void entryEvicted(EntryEvent<String, String> event) {
> > +                latch.countDown();
> > +            }
> > +        }, true);
> > +
> > +        Future<Void> future = map.setAsync("key", "value1", 3, TimeUnit.SECONDS);
> > +        future.get();
> > +        assertEquals("value1", map.get("key"));
> > +
> > +        assertTrue(latch.await(10, TimeUnit.SECONDS));
>
> We have a better method for this; HazelcastTestSupport.assertOpenEventually
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hazelcast/hazelcast/pull/7129/files#r48526150>.
>